src/client.ts

```diff
@@ -235,6 +239,32 @@ export default class Client implements ClientInterface {
     this.#ruby = ruby;
   }
 
+  async determineFormatter() {
+    const configuration = vscode.workspace.getConfiguration("rubyLsp");
+    const configuredFormatter: string = configuration.get("formatter")!;
+    switch (configuredFormatter) {
+      case "auto":
+        if (await this.projectHasDependency("rubocop")) {
+          this.client!.clientOptions.initializationOptions.formatter =
+            "rubocop";
+          this.client!.restart();
```

Comment:
@vinistock I'm wondering if there's a better way to handle this. Since the `determineFormatter` run asynchronously after the initialization, I need to load the updated config by restarting.

An alternative might be to have `determineFormatter` run synchronously but that would delay the initial startup time.

Reply:
Sorry, I don't understand why you need to restart. We already have a restart watcher if any Ruby LSP related configurations are changed. Is it not restarting when you change the config?